text/000-2018-2019-roadmap.md

```diff
@@ -0,0 +1,88 @@
+
+## Motivation
+
+This document is a distillation of multiple sources:
+
+1. The 2019 Community Survey.
+2. Community #EmberJS2019 blog posts, authored in response to our call for posts.
+3. Discussion on https://discuss.emberjs.com, Discord, Twitter, and other public venues.
+4. Deliberations among the Ember core teams.
+
+The goal of the RFC is to align the Ember community around a set of shared, achievable goals that balance the needs of existing users with the need to grow and support new users.
+
+## Detailed design
+
+Now that all of Ember Octane’s features are code complete, it’s time to turn our attention to what we’ll focus on over the next year.
+
+In this roadmap, our goal is to build on last year’s momentum and capitalize on the cutting-edge foundation we laid with Ember Octane.
+
+- **Continue simplifying Ember.** With many new APIs introduced in Octane, our focus this year should be removing features that feel out of date. The overall API surface area of Ember will shrink substantially, making it smaller, faster, and more approachable to new learners.
+- **Modernize our build system.** This year, we will put the finishing touches on Embroider and make it the default build pipeline for new applications, unlocking optimizations like tree-shaking and route-based code splitting.
+- **Better a11y by default**. We will fix the router to work out-of-the-box with assistive technologies, and give developers more tools for understanding and improving the accessibility of their Ember applications.
+- **Optimize for growth.** A feeling of momentum is critical for keeping the Ember community healthy. We will double down on evangelizing Ember Octane and making Ember approachable to a new generation of web developers.
+
+### Continue simplifying Ember
+
+Ember Octane put us on a strong footing, modernizing the most foundational APIs in Ember. We will build on that foundation to dramatically simplify the framework, **reducing the overall number of concepts** so that Ember is easier to learn, use, and debug.
+
+We will **extract or deprecate functionality that is no longer widely needed**. With `get()` and `set()` mostly gone, so goes the need for many Ember-specific utility functions. Some utilities can be moved into optional packages for those who need them, while others may be better replaced by libraries like [Lodash](https://lodash.com/).
+
+Candidates for deprecation or extraction include:
+
+- **Object utilities** such as `isBlank`, `compare`, `trySet`, etc.
+- **String utilities** such as `camelize` and `dasherize`.
+- **Computed property macros** such as `alias`, `mapBy`, `notEmpty`, etc.
+- **Legacy object model features** such as observers, `setProperties`, `incrementProperty`, etc.
+
+Perhaps the longest-awaited simplification is **eliminating controllers** from the Ember programming model, which we will do this year. This will require us to find a new home for query parameters.
```

Comment:
Wondering if dynamic query params is something that could be explored as part of these changes!

Reply:
Dynamic route segments also! e.g. handling both `foo.com/posts/:seo/:id` and `foo.com/posts/:id` with the same route handler.